src/unix/linux.c

```diff
@@ -1681,7 +1681,10 @@ int uv_cpu_info(uv_cpu_info_t** ci, int* count) {
     return UV__ERR(errno);
   }
 
-  fgets(buf, sizeof(buf), fp);  /* Skip first line. */
+  if (fgets(buf, sizeof(buf), fp) == NULL) {
+    fclose(fp);
+    return UV_EINVAL;
```

Comment:
For consistency with the rest of the function, I'd assign `fp = NULL;` after the call to `fclose()`.

Reply:
Thanks for your comments. I resolved it with a new commit.

Reply:
There's a memory leak now, the `cpus` array isn't freed. Also applies to all the other early returns.

That aside, we're reading from a /proc pseudo-file where reads can't fail so this is all pointless ceremony to appease an overzealous compiler. If you really want to get rid of the warnings, change the fread() calls to this:
```c
if (NULL == fgets(...))
  abort();
```
(and note the yoda style, I'm particular about that)

Reply:
I fixed it with a new commit. Just learn the style has a name. :)